src/client/Microsoft.Identity.Client/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -170,6 +191,19 @@ internal void InitCacheKey()
                 _credentialDescriptor = StorageJsonValues.CredentialTypeAccessTokenWithAuthScheme;
             }
 
+            if (AdditionalCacheKeyComponents != null)
+            {
+                _credentialDescriptor = StorageJsonValues.CredentialTypeAccessTokenExtended;
+                if (_extraKeyParts != null)
+                {
+                    _extraKeyParts.Concat(new[] { CoreHelpers.ComputeAccessTokenExtCacheKey(AdditionalCacheKeyComponents) });

```

Comment:
This is not covered by unit tests. YOu need a test where you combine POP with extra cache and validate the cache keys.